modules/k8s-cluster/main.tf

```diff
@@ -2,157 +2,118 @@ data "aws_caller_identity" "current" {}
 
 data "aws_region" "current" {}
 
-module "common" {
-  source = "../common-ignition"
-}
-
-data "aws_ami" "coreos" {
-  most_recent = true
-  owners      = ["595879546273"]
+resource "aws_eks_cluster" "eks-cluster" {
+  name     = "${var.cluster_name}"
+  role_arn = "${aws_iam_role.eks-cluster.arn}"
 
-  filter {
-    name   = "architecture"
-    values = ["x86_64"]
+  vpc_config {
+    security_group_ids = ["${aws_security_group.controller.id}"]
+    subnet_ids         = ["${var.subnet_ids}"]
   }
 
-  filter {
-    name   = "virtualization-type"
-    values = ["hvm"]
-  }
-
-  filter {
-    name   = "name"
-    values = ["CoreOS-stable-*"]
-  }
+  depends_on = [
+    "aws_iam_role_policy_attachment.eks-cluster-policy",
+    "aws_iam_role_policy_attachment.eks-service-policy",
+  ]
 }
 
-resource "aws_launch_configuration" "controller" {
-  name_prefix       = "${var.cluster_name}-controller-"
-  image_id          = "${data.aws_ami.coreos.image_id}"
-  instance_type     = "${var.controller_instance_type}"
-  enable_monitoring = false
-
-  user_data = "${data.ignition_config.controller-actual.rendered}"
+resource "aws_key_pair" "eks" {
+  key_name   = "${var.cluster_name}"
+  public_key = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDP1p65o186ezp0tpDid1qaIbNUg3QZmHqRtDiGOuzndebqK50uMKJ9KUGCPGXVEK+S1ztzMuRy/NB/U59UKxOyx/M6/oKclRH58+YRVg+ftp4hDNE9HW5o3mWvGmgOLtlaCfrpwXbOgtrT0pRH1R0qqeX0J3hY6m8JQlH+cHYdj7e/HjJQSpKaKmyBakQU8wHjvX4yjtxBRLdoLcOVQapHZPs2iFU8sqYT3FIGHSf3lyF/j+I9gNxe/B1KsTZpp+FrQ9mve7uSruK4fS0FvjTVZ/eemMm4niuAn4KGdFzmbHU3bBV1MpS04d4xrVQzhI/tXXVS8ZXF9Xekg2xtDmLWvl4hn43nfJa+8RSUVoeBv1XQRasafkgmF+ysTJhgsNZC8jziiEk7YqZh+uqkNRNIpveZU3bU+6aOzisM9VCA/HSoIEHLYUswXAXRbi75JMIhAT0l/RBTYpqnboOdb+MXM2jbcbJ2xbUJcFfDrpMHt1wE1Y+sp0P0zGEx9LubWQhplYYgMmW56NcYyHQTSS+V3EXaoEki17Qgg3MSMIdrWgH0c9EBnVj0L0dIvDzdYzUpbav0DAKu9ElYXcjnbDzZLmEyfZf5pnSC+NMCW2CF6E8C+FvcJ9akP+C3IU+tn5cc4u8eG5XuXh2vChXAs7B6slzvLPjpgbofuwVdErCrbw== daniel.blair@digital.cabinet-office.gov.uk"
```

Comment:
This is unfortunate that the cloudformation requires us to assign a key pair and doesn't come with the ssm-agent by default :(

We will have to address this in a follow up ... but daniels magic key is fine while we are just playing in sandbox

Reply:
added story to backlog to resolve